### DISCUSSION TITLE
About the Milvus version 2.2.x

### DISCUSSION THREAD

**Discussion by ingale726** [milvus-io/milvus#22014] [Q&A and General discussion]
Can Milvus version 2.2.x retrieve 100 million data in 1 second? I haven't tried it yet. Maybe it's faster to ask questions

**Comment by yhmo** [comment]
Do you mean "get embedding data of 100M rows" or "ANN search topk items from 100M rows"?

"get embedding data of 100M rows" ----- the milvus need to read data from storage one by one and return the data by rpc. it is very slow, time cost could be even more than 1 hour.

"ANN search topk items from 100M rows" ----- ANN search is fast, with proper cluster deployment and index/search parameters, each request latency could be even less than 100ms.

**Reply by ingale726** [reply]
I mean insert 100 million pieces of data and retrieve top10000.  I know the Milvus 2. x version can only use CPU, so I'm not sure if its performance can reach 100 million pieces of data, 1 second retrieval

**Reply by yhmo** [reply]
If the vector dimension is not high(<=1024), data is fully indexed, with multiple query nodes, we can easily reach the target(get top10000 from 100M within 1 second).

**Reply by ingale726** [reply]
Thanks
